app/models/ability.rb

```diff
@@ -0,0 +1,18 @@
+class Ability
+  include CanCan::Ability
+
+  def initialize(user)
+    if user.present?
+      if user.admin?
+        can :manage, :all
+      else
+        can %i(create read update destroy), [Journal, JournalEntry]
+        can %i(create show new), User
+        can :update, User, id: user.id
+        cannot :index, User
+      end
+    else
+      can %i(create show new), User
```

Comment:
How is this working? I can see when no user is present, we are letting them **create show new** users. Is that correct?

Reply:
I took out the show. But yeah, a new User has to create a User to enter the Application.